src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1205,14 +1205,30 @@ def _validate_ssh_key_helper(ssh_key_value, should_generate_ssh_keys):
 
 
 def _validate_vm_vmss_msi(cmd, namespace, from_set_command=False):
+
+    # For the creation of VM and VMSS, "--role" and "--scope" should be passed in at the same time
+    # when assigning a role to the managed identity
+    if not from_set_command and namespace.assign_identity is not None:
+        if namespace.identity_scope and not namespace.identity_role:
+            raise ArgumentUsageError(
+                "usage error: '--scope {}' is not applicable as the '--role' is not provided".format(
+                    namespace.identity_scope))
```

Comment:
Together with L1232, for simplicity, why don't we say:

> Usage error: To create role assignments, specify both --role and --scopes.

just like https://github.com/Azure/azure-cli/pull/21323

Reply:
In fact, I wanted to be consistent with the error message that the `--scope` parameter was not passed in the original code https://github.com/Azure/azure-cli/pull/21474/files#r826612256